Tests/BugsnagSwiftConfigurationTests.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  BugsnagSwiftConfigurationTests.swift
+//  Tests
+//
+//  Created by Robin Macharg on 22/01/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+import XCTest
+
+class BugsnagSwiftConfigurationTests: XCTestCase {
+
+    /**
+     * Objective C trailing-NSError* initializers are translated into throwing
+     * Swift methods, allowing us to fail gracefully (at the expense of a more-explicit
+     * (read: longer) ObjC invocation).
+     */
+    func testDesignatedInitializerHasCorrectNS_SWIFT_NAME() {
+        
+        do {
+            let _ = try BugsnagConfiguration(DUMMY_APIKEY_16CHAR)
+        }
+        catch let e as NSError {
+            XCTAssertEqual(e.domain, BSGConfigurationErrorDomain)
+            XCTAssertEqual(e.code, BSGConfigurationErrorCode.invalidApiKey.rawValue)
```

Comment:
In the case that the expected values for `BSGConfigurationErrorDomain` and `BSGConfigurationErrorCode` are inadvertently changed, this test would still pass. Testing for the expected literal values here would catch that.

```suggestion
            XCTAssertEqual(e.domain, "com.Bugsnag.CocoaNotifier.Configuration")
            XCTAssertEqual(e.code, 0)
```

Reply:
Good comment.  The thinking was that a user can incorporate the constants in their own use of the library, in which case equality checks (regardless of the actual values) would still work for them.  If we're testing that values of constants are set in stone I'd prefer to put that in a separate test to highlight the specific thing that may have changed.

Reply:
Ah I see. Exposing the constants is great for letting people determine what went wrong, and its nice to have them exposed. So maybe there should be tests for both things (that the constant names are correct and the constant values are what we expect)?

![porque no los dos](https://media.giphy.com/media/3o85xIO33l7RlmLR4I/giphy.gif)

Reply:
Win-win.  At the risk of overkill we now have both tests in.